File: Python/Program/RAM.py
from Komponen import Komponen

class RAM(Komponen):
    def __init__(self, kapasitas_GB=0, ddr="", merk="", nama=""):
        super().__init__(merk, nama)
        self._kapasitas_GB = kapasitas_GB
        self._ddr = ddr

    def set_kapasitas_GB(self, kapasitas_GB):
        self._kapasitas_GB = kapasitas_GB

    def set_ddr(self, ddr):
        self._ddr = ddr

    def get_kapasitas_GB(self):
        return self._kapasitas_GB

    def get_ddr(self):
        return self._ddr

File: CPP/Program/RAM.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class RAM : public Komponen{
	private:
		int kapasitas_GB;
		string ddr;
	
	public:
		RAM(){
			
		}
		
		RAM(int kapasitas_GB, string ddr, string merk, string nama) : 
		Komponen(merk, nama)
		{
			this->kapasitas_GB = kapasitas_GB;
			this->ddr = ddr;
		}
		
		void setKapasitasGB(int kapasitas_GB){
			this->kapasitas_GB = kapasitas_GB;
		}
		
		void setDdr(string ddr){
			this->ddr = ddr;
		}
		
		int getKapasitasGB(){
			return this->kapasitas_GB;
		}
		
		string getDdr(){
			return this->ddr;
		}
		
		~RAM(){
			
		}
};